### ISSUE TITLE
ObjectIds and UserIds paramter concerns

### ISSUE THREAD

**Issue by JeremyTBradshaw** [MicrosoftDocs/office-docs-powershell#4235]
I'd like to raise two concerns with these parameters.  The first is concerning this docs page.  The acceptable inputs should be clearly stated.  Instead all it says is this
> This parameter accepts a variety of objects, such as mailbox aliases, Send connector names, and so on.

But that just is not good enough.  The code behind the scenes is searching within specific properties, so why not just say what properties can be specified, as is the case for any other official Cmdlets' ID-related parameters.

The first point leads me to my second, which I don't actually expect to get resolved, so it's just a comment.  Why would the Search-AdminAuditLog cmdlet veer away from the standard style of Exchange Cmdlets and introduce "ObjectIds" and "UserIds" parameters in the first place?  If anything "Users" should refer to users, not admins (which I know in this case it may be non-admin users who ran commands that were audited).  It's just inconsistent.  Similar to the *-RecipientPermission cmdlets in Exchange Online, inconsistent.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4d36d45d-79cf-5ce9-0627-01bd4b8282c8
* Version Independent ID: 374a916d-f89d-99a3-c331-d9282c3825ed
* Content: [Search-AdminAuditLog (exchange)](https://docs.microsoft.com/en-us/powershell/module/exchange/policy-and-compliance-audit/search-adminauditlog?view=exchange-ps#feedback)
* Content Source: [exchange/exchange-ps/exchange/policy-and-compliance-audit/Search-AdminAuditLog.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/master/exchange/exchange-ps/exchange/policy-and-compliance-audit/Search-AdminAuditLog.md)
* Service: **exchange-powershell**
* GitHub Login: @chrisda
* Microsoft Alias: **chrisda**

**Comment by chrisda** [comment]
Regarding your first point, the reason why the ObjectIDs parameter description is so vague is because it's so vast. The object in question could literally be anything in the Exchange universe (a mailbox, transport rule, connector, accepted domain, journal rule, transport config, organization config, public folder, etc.) and the acceptable value is **highly** dependent on the type of object. Realistically, you'll probably need to run Search-AdminAuditLog *without* using the ObjectIds filter to then see examples of the valid object identity syntax (**ObjectModified** property value) that you're looking for (for example, Name, Canonical DN, public folder identity, etc.). It just varies too much between different object types.

I did some testing and updated the whole topic. I also added a link in the Description section to [Administrator audit log structure](https://docs.microsoft.com/en-us/Exchange/policy-and-compliance/admin-audit-logging/log-structure).

Regarding your second point: highly unlikely the parameter names will ever change; all we can do is describe them clearly and accurately.

Check out my updates and see if they'll do.
